code-challenges.test.js

```diff
@@ -15,37 +15,129 @@
+// })
+
+// Good Failure
+
+// const votes1 = { upVotes: 13, downVotes: 2 }
+// // Expected output: 11
+// const votes2 = { upVotes: 2, downVotes: 33 }
+// // Expected output: -31
+
+// // b) Create the function that makes the test pass.
+
+// First create a new function called "votechecker"
+// Create a variable called "upVotes" that will take in the value of the key "upVotes"
+// Create another variable called "downvotes" that will take the value of the key "downVotes"
+// Create another variable called "totalVotes" that will subtract the values of the upVotes and the downVotes
+// return our variable "totalVotes"
+
+// const voteChecker = (object) => {
+//   var upVotes = object.upVotes || 0
+//   var downVotes = object.downVotes || 0
+//   var totalVotes = upVotes - downVotes
+//   return totalVotes
+// }
+
+// Test passed
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+// First create a describe stament that contains the function "arrayCombo"
+// Create an it statement that explains how the function will take in two arrays and combine them with any duplicate values
+// Create an expect statement that take in the variables provided as arguments that will return our expected outcome
+
+// describe ("arrayCombo", () => {
+//   it("will take in two arrays as a parameter and combine them without any duplicate values", () => {
+//       expect(arrayCombo(dataArray1,dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"]
+//       )
+      
+      
+//   })
+// })
+
+// Good failure
+
+// const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+// const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+// // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
+// // b) Create the function that makes the test pass.
+
+// First create a new function called "arrayCombo" that takes in multiple arrays and parameters using the spread operator
+// Next create a variable called "joinedArray" that will combine the two arrays provided and store them into a different array
+// Next create another variable called "finalArray" that will remove the duplicated by converting the array into a set and then return the items (without duplicates) back into an array using the "Array.from" method
+// return the "finalArray"
+
+// const arrayCombo = (...array) => {
+//   var joinedArray = [].concat(...array)
+//   var finalArray = Array.from(new Set(joinedArray))
+//   return finalArray
+// }
+
```

Comment:
Very nice work with the stretch challenge! Great use of `Array.from()` and `new Set()`! To answer your question, there are many different ways to solve any particular problem in coding! This is very similar to how I would personally go about this challenge, but if you wanted to challenge yourself, try to solve it in a different way! A hint for a different method would be to try using the `.filter()` method to sort out any duplicates.